### ISSUE TITLE
ClientSide Conflict resolving not working as expected (Two Sync Trick)

### ISSUE THREAD

**Issue by Nobody84** [Mimetis/Dotmim.Sync#581]
Hi @Mimetis ,
I'm afraid I need your help again.

This time i tryed to resolve the conflict on the ClientSide unsing the _Two Sync Trick_ described in the [documentation](https://dotmimsync.readthedocs.io/Conflict.html#handling-conflicts-from-the-client-side).

### Environment
NetCore (3.1.0)
AspNetCore (3.1.10)

Dotmim.Sync.Core (0.8.0)
Dotmim.Sync.SqlServer (0.8.0)
Dotmim.Sync.SqlServer.ChangeTracking (0.8.0)
Dotmim.Sync.Web.Server (0.8.0)
Dotmim.Sync.Web.Sqlite (0.8.0)

Microsoft.EntitifyFramework (5.0.8)

### Reproduction steps:
I set the `ConflictResolutionPolicy` to  `ConflictResolutionPolicy.ServerWins` on the server and the client side.

1.Create a new row on the SqlServer and perform a synchronization `syncAgent.SynchronizeAsync()` to make sure both databases contains the the same row.
2. Make a modification to the `Text` and `LastModified` collumn of the row on both databases.
3. Performe the first synchronization `syncAgent.SynchronizeAsync()` and handle the conflict in the `syncAgent.LocalOrchestrator.OnApplyChangesFailed` method, and set the `SenderScopeId=null`.
```
syncAgent.LocalOrchestrator.OnApplyChangesFailed(acf =>
{
  // Check conflict is correctly set
  var localRow = acf.Conflict.LocalRow;
  var remoteRow = acf.Conflict.RemoteRow;
                        
  remoteRow["Text"] = remoteRow["Text"];
  remoteRow["LastModified"] = remoteRow["LastModified"];

  acf.SenderScopeId = null;
```
4. Performing the second synchronization.

### Result
The conflict has been resolved, but the rows did not change on both databases.
Here the output of the two syncs
```
### First Sync
Synchronization done.
        Total changes  uploaded: 1
        Total changes  downloaded: 1
        Total changes  applied: 1
        Total resolved conflicts: 1
        Total duration :00.00:00.058

### Second Sync
Synchronization done.
        Total changes  uploaded: 0
        Total changes  downloaded: 0
        Total changes  applied: 0
        Total resolved conflicts: 0
        Total duration :00.00:00.050
```

### Hints
Never the less I see, that my conflict resolving approach is a little bit different from the example shown in the documentation so I modified my approach and replaced the `remoteRow["Text"]` with "BlaBlaBla". Surprise surprise, this works. Further I diged deeper into it and recognized that the second sync only uploads data if at least one value of the remoteRow differst from the localRow (in the context of the `OnApplyChangesFailed` method. I think that, in the case both rows are equal, the second sync did not recognize any changes to the local row and so did not upload the changes.

### Example project
I have updated my example project und pushed it to a new branch: [https://github.com/Nobody84/DotmimSyncIssue/tree/issue-resolveconflictonclientside](https://github.com/Nobody84/DotmimSyncIssue/tree/issue-resolveconflictonclientside)
**Note** that the SqlServer connection strings must be modified in the WebApi/appsettings,json and DotmimSyncIssue/Program.cs

The program runs through three iteratrions.
1. The client wins, but I consider only the change to the `Text` value. => OK
2. The server wins, but I consider only the change to the `Text` value. => OK
3. The client wins, butnow  I consider the change to the `Text` and the `LastModified` value. => Not OK

Here ist the ouput of the program: [https://raw.githubusercontent.com/Nobody84/DotmimSyncIssue/issue-resolveconflictonclientside/output.txt](https://raw.githubusercontent.com/Nobody84/DotmimSyncIssue/issue-resolveconflictonclientside/output.txt)

**Comment by Mimetis** [comment]
I'm working on it.
In your first case, the only dif is that you don't change anything on the client side row 
the second case, you changed the last date. 
It seems the trigger is validated only at the end of the transaction.
Then on next sync the row is view as "too old" for .. a reason i don't understand yet

**Comment by Mimetis** [comment]
Hey @Nobody84 

First of all, thanks a lot for the time you took to create a sample repro.
I used it to debug and found the issue.

I know it's really time consuming create a sample repro project.
So.. thank you :)

Regarding the issue, I think I found the bug. 
The update table routine, on sqlite, does not update a tracking timestamp I need (explicitly for this scenario)

That being said, i DONT KNOW why ... it's working on the first scenario (with only one column updated)

Anyway, I should maybe make more research on WHY, but since I'm currently in hollidays, let's say that having a fix will be enough for now :)

Can you test the last commit on the master branch and see if it's resolving your issue ?

**Comment by Nobody84** [comment]
Hey @Mimetis 

Thanx for the quick response. I know for sure how frustrating it can be to get a bug report without having the ability to reproduce the error. So creating an sample, if possible, is the first thing I do if I think I found a bug. In some cases I recognized it wasn't even a bug, but I make mistakes in my production code. Don't worry about the time I used to build the sample. It would take me even more time to build my own sync agend ;).

I will test the master branch tomorrow and report the results.
